api-get-policy.go

```diff
@@ -35,7 +35,8 @@ func (c Client) GetBucketPolicy(bucketName, objectPrefix string) (bucketPolicy p
 		return policy.BucketPolicyNone, err
 	}
 	policyInfo, err := c.getBucketPolicy(bucketName)
-	if err != nil {
+	errResponse := ToErrorResponse(err)
+	if err != nil && errResponse.Code != "NoSuchBucketPolicy" {
```

Comment:
I am not sure how this change fixes #664 ? we are not returning any error when error is NoSuchBucketPolicy